### PR TITLE
addmv decomp (#94617)

### LINKED ISSUE #94617: Write a decomposition for addmv
### 🐛 Describe the bug

The operator is

```
- func: addmv.out(Tensor self, Tensor mat, Tensor vec, *, Scalar beta=1, Scalar alpha=1, Tensor(a!) out) -> Tensor(a!)
  structured: True
  dispatch:
    CPU: addmv_out_cpu
    CUDA: addmv_out_cuda
    MPS: addmv_out_mps
    SparseCsrCPU: addmv_out_sparse_compressed
    SparseCsrCUDA: addmv_out_sparse_compressed_cuda
```

Look in `torch/_decomp/decompositions.py` for examples. You can decompose it according to the formula in the documentation. However, this may not correctly handle stride behavior; talk to your bootcamp mentor if it does not.

It may be necessary to ensure this decomp is disabled for inductor (which currently does not compile matrix multiplies).

Once you're done python test/test_proxy_tensor.py -k addmv should xpass. Remove the xfail for addmv. Other tests may xpass too

### Versions

master

cc @SherlockNoMad @ngimel

### PR DESCRIPTION
**Pull request by cybershiptrooper**
Fixes #94617

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1136,6 +1136,19 @@ def addmm(self: Tensor, mat1: Tensor, mat2: Tensor, beta: int = 1, alpha: int =
     return out + beta * self
 
 
+@register_decomposition(aten.addmv)
+@out_wrapper()
+@pw_cast_for_opmath
+def addmv(self: Tensor, mat1: Tensor, vec: Tensor, beta: int = 1, alpha: int = 1):
```
Comment (vadimkantorov) [inline torch/_decomp/decompositions.py:1142]: is it correct to annotate here beta/alpha always as int?

Comment (cybershiptrooper) [reply torch/_decomp/decompositions.py:1142]: Hi, seems to be in here as well: https://github.com/pytorch/pytorch/pull/76621/files#diff-d25ad317caf41c3fdf1de3bcfc730102c9e4a3e68639559017ae45cb0bc26911R741

It didn't seem right but I thought it was some convention.. Should I remove this as well?

**Review by ngimel — APPROVED** [review]

**Comment by cybershiptrooper** [comment]
I'm very new to open source
If you were tagged, really sorry, everyone :")

**Pull request closed without merging**